presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationPushDown.java

```diff
@@ -113,70 +121,124 @@ public void testOffsetPushDown() throws ExecutionException, InterruptedException
+        RecordMessage recordMessage = createTimestampTestMessages(topicNameLogAppend);
         DistributedQueryRunner queryRunner = (DistributedQueryRunner) getQueryRunner();
         // ">= startTime" insure including index 2, "< endTime"  insure excluding index 4;
-        String sql = String.format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'",
-                topicNameLogAppend, timePair.first(), timePair.second());
+        String sql = format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'",
+                topicNameLogAppend, recordMessage.getStartTime(), recordMessage.getEndTime());
         ResultWithQueryId<MaterializedResult> queryResult = queryRunner.executeWithQueryId(getSession(), sql);
-        assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), 2);
+
+        String dumpMsg = buildDebugDumpString(topicNameLogAppend, recordMessage, queryRunner, sql, queryResult);
+        assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
+                .describedAs("with dump:\n%s", dumpMsg)
+                .isEqualTo(2);
+    }
+
+    private String buildDebugDumpString(String topic, RecordMessage recordMessage, DistributedQueryRunner queryRunner, String sql, ResultWithQueryId<MaterializedResult> queryResult)
+    {
+        StringBuilder sb = new StringBuilder();
+        // dump main messages
+        sb.append("Main messages:").append("\n");
+        for (int i = 0; i < recordMessage.getTestMessageSignatures().size(); i++) {
+            String dumpMessage = recordMessage.getTestMessageSignatures().get(i);
+            sb.append("  ").append(dumpMessage);
+            if (i == TIMESTAMP_TEST_START_INDEX) {
+                sb.append(", startTime:").append(recordMessage.getStartTime());
+            }
+            else if (i == TIMESTAMP_TEST_END_INDEX) {
+                sb.append(", endTime:").append(recordMessage.getEndTime());
+            }
+            sb.append("\n");
+        }
+
+        // dump sql and result
+        sb.append(format("dump test sql:%s\n", sql));
+        sb.append("dump test sql result:").append("\n").append(dumpSqlResult(queryResult.getResult()));
+
+        // dump data in kafka
+        sql = format("SELECT _partition_id,_partition_offset,_timestamp FROM default.%s WHERE _partition_offset between %s and %s order by _partition_id, _timestamp",
+                topic,
+                recordMessage.getStartOffset(),
+                recordMessage.getStartOffset() + MESSAGE_NUM);
+        MaterializedResult rows = queryRunner.execute(getSession(), sql);
+        sb.append("dump data check result:").append("\n").append(dumpSqlResult(rows)).append("\n");
+        return sb.toString();
+    }
+
+    private String dumpSqlResult(MaterializedResult rows)
```

Comment:
I suggested to inline this one. If you insist on keeping it separate please rename to  `buildResultsDebugDumpString`